### PR TITLE
Add JPEG2000 support

### PR DESCRIPTION
**Pull request by jcupitt**
Add JPEG2000 load and save with OpenJPEG.

### CODE REVIEW

File: libvips/include/vips/foreign.h [hunk 1]
```diff
@@ -676,6 +676,19 @@ int vips_niftiload_source( VipsSource *source, VipsImage **out, ... )
...
 
+int vips_jp2kload( const char *filename, VipsImage **out, ... )
+	__attribute__((sentinel));
+int vips_jp2kload_buffer( void *buf, size_t len, VipsImage **out, ... )
+	__attribute__((sentinel));
+int vips_jp2kload_source( VipsSource *source, VipsImage **out, ... )
+	__attribute__((sentinel));
+int vips_jp2ksave( VipsImage *in, const char *filename, ... )
+	__attribute__((sentinel));
+int vips_heifsave_buffer( VipsImage *in, void **buf, size_t *len, ... )
```
Comment (kleisauke) [inline libvips/include/vips/foreign.h:687]:
```suggestion
int vips_jp2ksave_buffer( VipsImage *in, void **buf, size_t *len, ... )
```
(should fix the failing CI build, I think)

Comment (jcupitt) [reply libvips/include/vips/foreign.h]: why is copy-paste so hard :(

File: libvips/foreign/jp2ksave.c [hunk 2]
```diff
@@ -0,0 +1,1181 @@
...
+		vips_error( class->nickname,
+			"%s", _( "not an integer format" ) );
+		return( -1 );
+	}
+
+	switch( jp2k->subsample_mode ) {
+	case VIPS_FOREIGN_SUBSAMPLE_AUTO:
+		jp2k->downsample =
+			!jp2k->lossless &&
+			jp2k->Q < 90 &&
```
Comment (lovell) [inline libvips/foreign/jp2ksave.c:603]: Should this be `<= 90` to match the docs?

Comment (jcupitt) [reply libvips/foreign/jp2ksave.c:603]: I'm so confused, I though 90 and over was no chroma subsampling, 89 was chroma subsampled. Let me check.

Comment (jcupitt) [reply libvips/foreign/jp2ksave.c:603]: OK, made it >= 90 everywhere. The docs, comments and code were all inconsistent :(

**Pull request merged**